sparkmagic/sparkmagic/tests/test_livysession.py

```diff
@@ -376,6 +376,9 @@ def test_logs_gets_latest_logs(self):
     def test_wait_for_idle_returns_when_in_state(self):
         self.http_client.post_session.return_value = self.session_create_json
         self.get_session_responses = [self.ready_sessions_json,
+                                      self.ready_sessions_json,
```

Comment:
I don't think that all these unit tests should be actually using the http_client when retrieving the html table for the session. Maybe make the `get_session_row_html` part of session and simply use a `MagicMock` in every unit test and just verify that it's called?